packages/dcos-integration-test/extra/test_remove_dockercfg_hook.py

```diff
@@ -0,0 +1,36 @@
+import uuid
+
+import pytest
+
+
+@pytest.fixture(scope='module')
+def auth_cluster(cluster):
+    if not cluster.dockercfg_hook_enabled:
+        pytest.skip("Skipped because not running against cluster with remove .dockercfg hook.")
+    return cluster
+
+
+def test_remove_dockercfg_hook(cluster):
```

Comment:
This test is supposed to verify that the dockercfg is not exposed inside of the container, when using the fetcher to inject dockercfg's for talking to private registries. We added a special hook to remove the dockercfg from the sandbox before the container starts. Here's the intended flow of this test:
1. On setup, drop a `.dockercfg` on every node's filesystem, in particular targeting the agents. This could just as well be fetched from an http/ftp URL.
2. Metronome asks Mesos to launch a task with the dockercfg's URI as a fetcher artifact
3. Mesos fetcher copies the dockercfg from the `file://` location into the sandbox prepared for the task
4. If this task was a docker image (which it's not), the container runtime (UCR or Docker) would use the dockercfg's credentials to pull the image, e.g. from a private registry
5. Then the hook swoops in and deletes the dockercfg from the sandbox directory (although it still lives in the original location from whence it was fetched)
6. Then the container is launched and the task is run, which checks for the existence of the dockercfg file in the task's sandbox. The task is gone, so `test -f` returns a non-zero status code, which `metronome_one_off`'s `wait_for_completion` will detect as a failure, and raise an exception. I'm not sure why it doesn't. Maybe because we're never enabling the hook?
I think the job should actually `test ! -f` as was happening before, and we should set `DOCKERCFG_HOOK=true` in at least one CI installer configuration
I'll clean this up a bit and see if I can make it clearer in #1091 

Bonus points:
- Verify that a fetched file of a different name is not removed.
Double-bonus points:
- Use the .dockercfg to actually fetch an image from a private registry. Do we have any integration tests that use a private docker registry?